### DISCUSSION TITLE
Ezcoo 4K 120Hz VRR CEC HDMI 2.1 Splitter EZ-SP12H21

### DISCUSSION THREAD

**Discussion by awawa-dev** [awawa-dev/HyperHDR#1004] [Show and tell]
I'm curious about your experiences. I just received it and it turns out that CEC works (it's supposedly unidirectional). In the older EZ-SP12H2 model, CEC didn't work. In my case, RPI4 is connected to the splitter input and Output1 to the LG TV. HyperHDR recognizes commands when the TV is turned on/off and the LG remote control buttons.

**Comment by litui** [comment]
CEC works fine to output 1 on this splitter. Having taken my SP12H21 apart, done some probing, and microsoldered a lead to pin 13, I can tell you it's a straight-through connection between pin 13 on the input and pin 13 on output 1, so there's no directionality.

I've got cec-gpio working on my RPi 4 though sadly it doesn't seem to work with libcec in *bookworm* (only kernel-native cec-ctl). That's what prompted my efforts in #1005.

**Comment by awawa-dev** [comment]
I tested the Raspberry Pi bookworm OS and fortunately CEC still works. Good to know there is a direct connection between pins 13. The manufacturer's description indicated that they used a diode so it is unidirectional.

**Reply by litui** [reply]
Interesting. Next time I have it open I'll have to retest continuity. I admittedly was only testing in one direction so it's *possible* there's a diode.

**Reply by litui** [reply]
If there is a diode, it might actually explain some things I was seeing... or _wasn't_ seeing as the case may be, since I soldered my monitoring lead to pin 13 on output 1 rather than on the input... hmm.

**Reply by litui** [reply]
As for the bookworm issues I'm seeing that might just be related to using cec-gpio on the RPi since I don't have a Pulse-Eight dongle.

**Reply by awawa-dev** [reply]
EZ-SP12H21 description says *"Does not support Sonos-ARC/e-ARC CEC Bi-directional."* But maybe it's a mistake and they missed a "." between a "e-ARC" and "CEC"? ;-)

**Reply by litui** [reply]
Interestingly, I have a Sonos eARC setup, but it goes through my TV's eARC port. Seems to be working just fine...

**Reply by litui** [reply]
Confirmed: there is **not** a diode between the Input and Output 1 on pin 13. It **is** a direct connection (And there is no CEC on Output 2, as far as I can tell).

I took my SP12H21 apart again today, tested continuity on pin 13 in both directions, and soldered a lead to pin 13 on the input so I could confirm it all works by monitoring each pin using cec-ctl on the RPi. The below screenshot shows identical messages at both Input and Output.

<img width="1097" alt="Screenshot 2024-12-02 at 2 00 48 PM" src="https://github.com/user-attachments/assets/03c92fd3-b21f-4f9b-9317-6bfba8a977f4">

**Reply by awawa-dev** [reply]
So this is probably what they meant, that it is not bi-directional: there is no CEC on output 2. It should be obvious since the master device is always output 1, but they described it in such a way that it is not entirely clear what they mean.

**Comment by JFingerle** [comment]
I needed to solder a wire from input pin 13 to output (1) pin 13. The print on my pcb is "SP12H2 V0.4 2424".
![cec_mod](https://github.com/user-attachments/assets/b3803d0b-6338-409a-bd62-423e124b0c46)

**Reply by Oli-70** [reply]
Can someone confirm that it works?

**Comment by gurpreetsingh00885** [comment]
@awawa-dev does this splitter also support LLDV like sp12h2?

**Reply by awawa-dev** [reply]
I don't remember if I tested this, and I don't use DV on a daily basis, maybe some other user can confirm it? I think most Ezcoo splitters/matrix enforce LLDV in the firmware, but personally I can't guarantee anything for something that even the official manufacturer doesn't boast about, and the next firmware version might break it.

**Reply by gurpreetsingh00885** [reply]
Thanks for the quick reply!

**Reply by lev-ginsburg** [reply]
I also just received this splitter, and it appears to be having issues with DV coming from a roku ultra (when in jellyfin/netflix apps). I also tried running calibration but hyperhdr wasn't able to pick up any of the patterns when the colors were skewed. The roku and TV both report being in DV mode as well.

It does appear that Dolby Vision is listed on the product specs for the sp12h21: https://www.easycoolav.com/products/4k-120hz-hdmi-21-splitter-8k-60hz-1-in2-out-vrr-allm

![Screenshot_20251104-122830](https://github.com/user-attachments/assets/edd425fa-6950-4e59-ae89-7d07bf871035)

Hoping for any guidance that might resolve the color issues in this mode. Thanks!

**Reply by awawa-dev** [reply]
If we rule out unusual custom LUTs or grabber driver errors, which you'll learn more about if you read the threads, then it could be Dolby Vision of course (typical pink color), but not LLDV for sure. EDID is only a suggestion during HDMI connection negotiations; what the source device (or intermediate devices) does is a different matter.

**Reply by lev-ginsburg** [reply]
There shouldn't be any custom LUTs set as the calibration failed, but I'd be happy to reset to defaults if there is a way to do that in the UI. I can look into seeing if there are any driver errors being output from the info in the other threads.

**Reply by awawa-dev** [reply]
Check on Windows with the default installation. If it is indeed Dolby Vision using a different than LLDV protocol, as indicated by the pink color, no calibration will help you because it is a closed, proprietary format. Low Latency Dolby Vision can only be captured correctly because it is essentially HDR10 without metadata. Only some HDFury and Ezcoo devices can force LLDV using special EDID settings, and that only under favorable circumstances, meaning all devices agree to it.

**Reply by gurpreetsingh00885** [reply]
@lev-ginsburg what grabber do you use?
I am still waiting for my splitter.
Maybe you can try playing with the EDID and the scaling settings on the splitter?

And yes you can check using windows like awawa suggested.

Otherwise we will have to look for custom firmware I guess which may not even exist...

**Reply by lev-ginsburg** [reply]
I orderd an MS2130 from aliexpress: https://www.aliexpress.us/item/3256804804393973.html?spm=a2g0o.order_list.order_list_main.5.76781802FPPP5b&gatewayAdapt=glo2usa

I have tried changing the EDID switches on the ezcoo, setting it to both the 4k@120 DV and 4k@60 DV settings. Scaling the output from 4k to 1080 made no difference. These both register as 4k Dolby Vision in the roku display settings, however leaving the ezcoo at default settings (all 0s) makes the roku report a 4k TV (no HDR or DV). Will provide screenshots in a follow up post

**Reply by awawa-dev** [reply]
So you have connected/configured it wrong or something wrong is with the splitter:  EDID pass-through should work if you connected TV to Output 1(scaller off) and grabber to Output 2 (scaller on). Consult the HDMI splitter manual.

**Reply by gurpreetsingh00885** [reply]
I emailed support@easycoolav.com to ask about LLDV:

> Hi,
> 
> I wanted to know if this splitter can also force low latency dolby vision like the SP12H2.
> 
> Thanks,
> Gurpreet

Response from tech support:

> Yes, sure. 
> The only different is 4K120 and 4K60

**Reply by awawa-dev** [reply]
I wouldn't put much stock in such claims, as they either don't know what they're selling or, as is usually the case with companies like these, essentially pushing equipment from China, they're lying. Another significant difference is VRR and CEC support. Interestingly, the experimental MS2130 firmware can also force LLDV, though this should only be used for testing, as we won't be able to split the signal to a TV.

**Reply by gurpreetsingh00885** [reply]
@awawa-dev

Sorry this is a bit off-topic. I had an ms2131 which for some reason worked fine on 4k60 SDR content (talking about pass through; capture was 1080p60 and I don't really care) but soon as HDR was turned on, it would stop capturing (that old familiar grainy noise image like when old TVs lost signal). This was happening no matter how much I lowered the resolution.

I also have an ms2109 which was capturing 1080p60 HDR fine at least (and with DV the image had that expected pinkish hue to it).

Then I came across a discussion where you mentioned flashing ms2131 with your experimental F3 firmware for ms2130 and I took a risk and did that and while I lost the pass through, everything else started working way better. Even 1080p120 is being captured. And yes LLDV was also working with my fire stick.
Only problem I faced was that on my PS5, for 1080p HDR the supported resolutions didn't have 120fps. I am surprised this happened as the splitter can support 4k60 pass through and that must mean it is hdmi 2.0 but cannot support 1080p120fps HDR for my PS5 (I'm hoping the splitter will help with this maybe as I heard the SP12H2 downscales from 4:4:4 to 4:2:0 perhaps. otherwise I will try to tinker with the firmware and update the EDID on the ms2131 itself - hoping that it was the downgrade to ms2130 which makes my ps5 think it won't support 1080p120 HDR - I could be terribly wrong here though).
(I had another splitter which does support 4k120fps HDR but doesn't downscale so I was able to forcefully feed 1080p120 HDR into the capture card and the captured screen was just black, no flashing or anything).

Sorry for the long reply but I just wanted to thank you for that firmware file :)

**Reply by awawa-dev** [reply]
You can force 120Hz for ms2130/ms2131 only by the splitter ignoring their capabilities. Do not forget these grabbers are only HDMI1.4 devices. Some lower bandwidth encoding worked in my case: search the ms2130 topic.

**Reply by gurpreetsingh00885** [reply]
Yes I do understand that. Unfortunately the splitter I currently have cannot downscale and my ps5 won't use any other encoding than whatever it uses by default.
Waiting for this splitter and will test when I get it.
Thanks.

Since you said ms2131 is also HDMI1.4 - my capture card does support 4k60 for the passthrough. Google says it must be at least HDMI 2.0.

**Reply by awawa-dev** [reply]
> Since you said ms2131 is also HDMI1.4

not me, the official chip specification

**Reply by lev-ginsburg** [reply]
Just for completeness, I also ordered these cables to go from the roku -> splitter -> grabber: https://www.amazon.com/dp/B081D96HPV?ref=ppx_yo2ov_dt_b_fed_asin_title&th=1. They should be able to handle the bandwidth needed to send the DV signal all the way down

and I'm running a rpi 4 with the latest hyperhdr image on it

**Reply by awawa-dev** [reply]
I highly doubt that the cable has any involvement in this particular situation.

**Reply by AstaRom** [reply]
> Otherwise we will have to look for custom firmware I guess which may not even exist...

- [Alternative Devices for Enabling LLDV](https://www.avsforum.com/threads/alternative-devices-for-enabling-lldv-please-read-posts-1-2.3254266)
- [xnappo / ez_dv_edid_fw](https://github.com/xnappo/ez_dv_edid_fw)

**Reply by gurpreetsingh00885** [reply]
Fabulous!

**Reply by gurpreetsingh00885** [reply]
@awawa-dev I have this capture card: 

https://amzn.in/d/g19NFbQ

It supports 4k60 output to my TV when using my fire tv stick (confirmed myself) and macbook.

I also opened it up and the chip says it is ms2131.

4k60 must mean it is HDMI 2.0. I am trying to update the EDID for the firmware on my card to make 1080p 120 HDR show up on my PS5 (experimenting till I recieve SP12H21).
For that I believe I need 1080p120 HDR 4:4:4.
What I have tried is backup my firmware, open it in a hex editor, find the edid using the header and then copy 256 bytes. Convert and save into a .bin file. Then open that in AWEDID editor and modify stuff in there. Then I dump the EDID to the backed up firmware using a tool called ms213x-rename that I found on GitHub. 
However that firmware breaks the capture card completely and I have to flash the clean backup again. 

Am I missing anything? 
Can you please be kind enough to share how you modify firmware?

Right now my goal is for 1080p HDR 120FPS to just show up for my ps5.

**Reply by awawa-dev** [reply]
http://en.macrosilicon.com/info.asp?base_id=2&third_id=76 I can see 1.4b bandwidth support only. And 4:4:4 is the most bandwidth hungry encoding probably way to much for 1080p120 and this grabber. Firmware needs signing/CRC and I've done it almost 2 years ago and since the result was satisfying I'm done with it.

**Comment by romquenin** [comment]
Hello everyone, I'm having a problem with this splitter and a PI4.
I connected "output 1/main" of my Onkyo amplifier to the splitter's input, output 1 to my LG OLED TV, and output 2 to the MS2130 capture card.
All three EDID levers are in the up position (copy mode), and the lever for output 2 is down (scaler on). With this setup, my TV only displays this strange, grainy image. I'm using only 2.1 HDMI Cables.
<img width="530" height="336" alt="shot181809" src="https://github.com/user-attachments/assets/6057fd3b-537d-4818-a2da-67ce17932a45" />
If I swap the splitter's outputs, the TV and grabber displays the image correctly, but CEC doesn't work.
For now, the only solution I've found to get everything working more or less correctly is to connect the main HDMI output of my amplifier directly to my TV (CEC works) and the sub HDMI output of my amplifier to the splitter input or directly to the capture device. As such, in both cases, the HDR, LLDV, and VRR modes work correctly.
Using the splitter is therefore completely unnecessary, as I can't get 4K120 to 1080p120 to work (Option 120 does not appear in the grabber options on HyperHDR).
Do you have any idea what's wrong with my setup?

**Reply by romquenin** [reply]
I sent an email to support@easycoolav.com explaining my problem, as this bug is not actually related to HyperHDR.

**Comment by SimpleMichaelR** [comment]
Hey guys, are the Ezcoo and Rofavezco same? Wanna order 120hz splitter, but can't find exactly ezcoo, only rofavezco
https://www.amazon.com/Splitter-HDCP2-3-Vision-Scaler-Monitors/dp/B0CXXN58DJ/ref=sr_1_4

**Reply by awawa-dev** [reply]
These devices are often rebranded, so the safest option is to ask the Amazon seller directly.

**Reply by Oli-70** [reply]
Once you have asked, feel free to share your information please

**Reply by SimpleMichaelR** [reply]
So, today I got an answer from official support.

Hello Michael,
Yes, you can buy it on Amazon.
They are the same product, due to various reasons, we have used different names.


Merry Christmas and Happy New Year!

Best Regards

EZCOO. INC - Tech Support Team

support@easycoolav.com
